tests/samples/cmake-files/targetLinkLibrariesWithDynamicLookup.cmake

```diff
@@ -0,0 +1,278 @@
+#
+# - This module provides the function
+# target_link_libraries_with_dynamic_lookup which can be used to
+# "weakly" link loadable module.
+#
+# Link a library to a target such that the symbols are resolved at
+# run-time not link-time. This should be used when compiling a
+# loadable module when the symbols should be resolve from the run-time
+# environment where the module is loaded, and not a specific system
+# library.
+#
+# Specifically, for OSX it uses undefined dynamic_lookup. This is
+# similar to using "-shared" on Linux where undefined symbols are
+# ignored.
+#
+# Additionally, the linker is checked to see if it supports undefined
+# symbols when linking a shared library. If it does then the library
+# is not linked when specified with this function.
+#
+# http://blog.tim-smith.us/2015/09/python-extension-modules-os-x/
+#
+#
+# The following functions are defined:
+#
+#   has_dynamic_lookup(<ResultVar>)
+#
+# Check if the linker requires a command line flag to allow leaving symbols
+# unresolved until runtime.  A test project is built and ran and ``<ResultVar>``
+# is set based on whether the project was built and ran successfully.  The
+# result is cached between invocations and recomputed only when the value of
+# ``CMAKE_SHARED_LINKER_FLAGS`` changes.
+#
+# Defined variables:
+#
+# ``<ResultVar>``
+#   Whether the linker allows undefined symbols for shared libraries.
+#
+# ``HAS_DYNAMIC_LOOKUP``
+#   Cached alias.
+#
+#
+#   has_symbol_dedupe(<ResultVar>)
+#
+# Check if the linker/loader correctly coalesces symbols that have been
+# duplicated across link boundaries.  A test project is built and ran and
+# ``<ResultVar>`` is set based on whether the project was built and ran
+# successfully.  The result is cached between invocations and recomputed only
+# when the value of ``CMAKE_SHARED_LINKER_FLAGS`` changes.
+
+# Defined variables:
+#
+# ``<ResultVar>``
+#   Whether the linker/loader correctly coalesces duplicated symbols
+#
+# ``HAS_SYMBOL_DEDUPE``
+#   Cached alias.
+#
+
+function(_test_weak_link_project projectName testName)
+  set(options DYNAMIC_LOOKUP SYMBOL_DEDUPE)
+  cmake_parse_arguments(_args "${options}" "" "" ${ARGN})
+
+  set(test_project_src_dir
+      "${PROJECT_BINARY_DIR}/CMakeTmp/${projectName}/src")
+  set(test_project_bin_dir
+      "${PROJECT_BINARY_DIR}/CMakeTmp/${projectName}/build")
+
+  file(MAKE_DIRECTORY ${test_project_src_dir})
+  file(MAKE_DIRECTORY ${test_project_bin_dir})
```

Comment:
Could you include the description of the test project ? 

May be by adapting/copying part of the description that is available here: https://github.com/opadron/weak-linking-demo ?